vhost-device-vsock/src/main.rs

```diff
@@ -1045,6 +1045,8 @@ mod tests {
         test_dir.close().unwrap();
     }
 
+    // TODO: Remove the ignore attribute when CI gets fixed
+    #[ignore]
```

Comment:
What about a runtime check, so we can run these tests at least locally?

I mean something like this:

```diff
--- a/vhost-device-vsock/src/vhu_vsock_thread.rs
+++ b/vhost-device-vsock/src/vhu_vsock_thread.rs
@@ -1038,11 +1038,15 @@ mod tests {
         test_dir.close().unwrap();
     }
 
-    // TODO: Remove the ignore attribute when CI gets fixed
-    #[ignore]
     #[cfg(feature = "proxy")]
     #[test]
     fn test_vsock_thread_vsock_backend() {
+        if VsockListener::bind_with_cid_port(libc::VMADDR_CID_LOCAL, libc::VMADDR_PORT_ANY).is_err()
+        {
+            println!("  SKIPPED: AF_VSOCK is not available.");
+            return;
+        }
+
         let groups: Vec<String> = vec![String::from("default")];
         let cid_map: Arc<RwLock<CidMap>> = Arc::new(RwLock::new(HashMap::new()));
```

Tested in this way:

```
$ cargo test -p vhost-device-vsock -- --nocapture
...
$ sudo rmmod vsock_loopback 
$ cargo test -p vhost-device-vsock -- --nocapture | grep SKIPPED
...
  SKIPPED: AF_VSOCK is not available.
```

Reply:
For this test it makes sense to ignore during execution. I explicitly ignored all 4 because there is a problem in the main.rs test the one that is ignored. The test itself doesn't do AF_VSOCK connection but causes the vsock backend code to establish a vsock connection. What should I do about those?

Reply:
If ` VsockListener::bind_with_cid_port(libc::VMADDR_CID_LOCAL, libc::VMADDR_PORT_ANY)` works, I'm expecting that also making a connection works, so that should be a "generic" way to probe that AF_VSOCK is supported by the host. So IMHO we can use the same probe at the beginning of all tests where we know we will use AF_VSOCK for listening or connecting.

Maybe I didn't understand the question correctly, but does it solve your doubts?

To be clear, I would apply this diff to all tests where we put ignore.

Reply:
Sorry I had misread the diff. It should work. Will fix. Thanks!

Reply:
This is a good suggestion. Thanks! Done.